File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/ExtremismProfile.java
package uk.gov.justice.digital.hmpps.riskprofiler.model;

import io.swagger.annotations.ApiModelProperty;
import lombok.*;

import javax.validation.constraints.NotBlank;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@ToString
public class ExtremismProfile extends RiskProfile {
    @ApiModelProperty(value = "Indicates Regional CT Lead should be informed", example = "false", position = 3)
    private boolean notifyRegionalCTLead;
    @ApiModelProperty(value = "Indicates that there is data to indicate that this person has an increased risk of engaging in extremism", example = "false", position = 4)
    private boolean increasedRiskOfExtremism;

    public RiskType getRiskType() {
        return RiskType.EXTREMISM;
    }

    @Builder(builderMethodName = "extremismBuilder")
    public ExtremismProfile(@NotBlank String nomsId, @NotBlank String provisionalCategorisation, boolean notifyRegionalCTLead, boolean increasedRiskOfExtremism) {
        super(nomsId, provisionalCategorisation);
        this.notifyRegionalCTLead = notifyRegionalCTLead;
        this.increasedRiskOfExtremism = increasedRiskOfExtremism;
    }
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/model/ExtremismProfile.kt
package uk.gov.justice.digital.hmpps.riskprofiler.model

import io.swagger.v3.oas.annotations.media.Schema
import javax.validation.constraints.NotBlank

data class ExtremismProfile(
  val nomsId: @NotBlank String,
  var provisionalCategorisation: @NotBlank String,
  @field:Schema(title = "Indicates Regional CT Lead should be informed", example = "false")
  var notifyRegionalCTLead: Boolean = false,
  @field:Schema(
    title = "Indicates that there is data to indicate that this person has an increased risk of engaging in extremism",
    example = "false"
  )
  var increasedRiskOfExtremism: Boolean = false
) : RiskProfile(nomsId, provisionalCategorisation) {

  constructor() : this("", "", false, false)

  override fun getRiskType(): RiskType {
    return RiskType.EXTREMISM
  }
}
